### PR TITLE
Expose GetASNInt as part of WOLFSSL_ASN_API

### PR DESCRIPTION
**Pull request by anhu**
Fixes ZD 16070

### CODE REVIEW

File: wolfssl/wolfcrypt/asn.h [hunk 1]
```diff
@@ -2194,6 +2194,11 @@ WOLFSSL_LOCAL int GetMyVersion(const byte* input, word32* inOutIdx,
 WOLFSSL_LOCAL int GetInt(mp_int* mpi, const byte* input, word32* inOutIdx,
                          word32 maxIdx);
 
+#ifndef WOLFSSL_ASN_TEMPLATE
+WOLFSSL_ASN_API int GetASNInt(const byte* input, word32* inOutIdx, int* len,
```
Comment (dgarske) [inline wolfssl/wolfcrypt/asn.h:2198]: Let's add a version for WOLFSSL_ASN_TEMPLATE. I think its similar to template's GetInt. @SparkiDev should help with this.

```
ASNGetData dataASN[intASN_Length];
XMEMSET(dataASN, 0, sizeof(dataASN));
ret = GetASN_Items(intASN, dataASN, intASN_Length, 0, input, inOutIdx, maxIdx);
```

**Review by dgarske — CHANGES_REQUESTED** [review]
Looks good. Any other API's that we should consider? Let's add an ASN template version too.

**Comment by SparkiDev** [comment]
The ASN_TEMPLATE version of GetASNInt can call GetASN_Integer() to share code.
Still needs to call GetASNHeader though.

**Comment by SparkiDev** [comment]
Leave it as is for WOLFSSL_ASN_TEMPLATE.
Alternatively, take a look at GetShortInt for an idea of how to implement.

File: wolfssl/wolfcrypt/asn.h (hunk 1 above)
Comment (anhu) [reply wolfssl/wolfcrypt/asn.h:2196]: I took @SparkiDev 's suggestion and just exposed `GetASNInt()` even when `ASN_TEMPLATE` is enabled.

**Review by dgarske — APPROVED** [review]

**Comment by anhu** [comment]
retest this please

**Comment by anhu** [comment]
retest this please

**Pull request merged**